src/transaction/transaction.ts

```diff
@@ -18,11 +18,14 @@ const TxSigned = (
   byronWitnesses: TxWitnessByron[],
   shelleyWitnesses: TxWitnessShelley[],
 ): SignedTxCborHex => {
-  const [txBody, meta] = unsignedTxDecoded
+  const { txBody, scriptWitnesses, meta } = unsignedTxDecoded
   const witnesses = new Map()
   if (shelleyWitnesses.length > 0) {
     witnesses.set(TxWitnessKeys.SHELLEY, shelleyWitnesses)
   }
+  if (scriptWitnesses && scriptWitnesses.length > 0) {
+    witnesses.set(TxWitnessKeys.NATIVE_SCRIPTS, scriptWitnesses)
```

Comment:
I think we should explicitly use "native" in the name of variables that deal with native scripts only.